src/test/scala/strawman/collection/test/TraverseTest.scala

```diff
@@ -128,12 +128,16 @@ class TraverseTest {
     val xs3: immutable.List[String] = xs2
 
     val xs4 = immutable.TreeMap((1, "1"), (2, "2"))
-    val xs5 = flatCollect(xs4) { case (2, v) => immutable.List((v, v)) }
+    val xs5 = flatCollect(xs4) { case (2, v) => immutable.List((v, v)) }(immutable.TreeMap)
```

Comment:
I created https://github.com/scala/bug/issues/10427 and https://github.com/lampepfl/dotty/issues/2911